src/DefaultCoreManager.cc

```diff
@@ -177,26 +177,28 @@ DefaultCoreManager::adjustCores() {
         // creation that just received an exclusive core, but such a race is
         // safe as long as it results only in the failure of the exclusive
         // thread creation.
-        for (uint32_t i = 0; i < exclusiveCores.size(); i++) {
-            int coreId = exclusiveCores[i];
-            MaskAndCount slotMap = *occupiedAndCount[coreId];
-            if (slotMap.numOccupied == maxThreadsPerCore - 1) {
-                // Attempt to reclaim this core with a CAS. Only move back to
-                // sharedCores if we succeed.
-                MaskAndCount oldSlotMap = slotMap;
-                slotMap.numOccupied = maxThreadsPerCore;
-                if (occupiedAndCount[coreId]->compare_exchange_strong(
-                        oldSlotMap, slotMap)) {
-                    exclusiveCores.remove(i);
-                    *lastTotalCollectionTime[coreId] = 0;
-                    *occupiedAndCount[coreId] = {0, 0};
-                    sharedCores.add(coreId);
-                    continue;
+        if (estimate == 1) {
```

Comment:
Please modify the previous condition instead. This extra nesting seems harder to understand.